src/lints/pub_function_changed_abi.ron

```diff
@@ -0,0 +1,58 @@
+SemverQuery(
+    id: "pub_function_changed_abi",
+    human_readable_name: "pub function changed ABI",
+    description: "A public function changed its external ABI.",
+    required_update: Major,
+    reference_link: Some("https://doc.rust-lang.org/reference/items/external-blocks.html#abi"),
+    query: r#"
+    {
+        CrateDiff {
+            baseline {
+                item {
+                    ... on Function {
+                        name @output
+                        visibility_limit @filter(op: "=", value: ["$public"]) @output
+
+                        importable_path {
+                            path @output @tag
+                        }
+                        
+                        span_: span @optional {
+                            filename @output
+                            begin_line @output
+                        }
+                        
+                        abi_: abi {
+                            name @output @tag
+                        }
+                    }
+                }
+            }
+            current {
+                item {
+                    ... on Function {
+                        visibility_limit @filter(op: "=", value: ["$public"])
+                        
+                        importable_path {
+                            path @filter(op: "=", value: ["%path"])
+                        }
+                        
+                        abi @fold @transform(op: "count")  @filter(op: "=", value: ["$zero"]) {
+                            name @filter(op: "=", value: ["%name"])
+                        }
+
+                        new_abi_: abi {
+                            name @output
+                        }
+                    }
+                }
+            }
+        }
+    }"#,
+    arguments: {
+        "public": "public",
+        "zero": 0,
+    },
+    error_message: "A publicly-visible function changed it's ABI type.",
+    per_result_error_template: Some("function {{join \"::\" path}} changed ABI type from {{abi_name}} to {{new_abi_name}} in {{span_filename}}:{{span_begin_line}}"),
```

Comment:
```suggestion
    per_result_error_template: Some("{{join \"::\" path}} changed ABI from {{abi_name}} to {{new_abi_name}} in {{span_filename}}:{{span_begin_line}}"),
```
We really want to try to fit this on one line in the terminal, so brevity is everything.